carbon-design-system/carbon
# refactor(InlineCheckbox): ariaLabel to aria-label

#12802 

Update `InlineCheckbox` to use the proper `aria-label`

#### Changelog

**New**

- n/a

**Changed**

1. added `deprecated()` to the old propType
2. added new usage for `aria-label`
3. kept backwards compatibility so that both aria versions can work with `ariaLabel` (deprecated) and `aria-label` (new).
4. updated tests API

**Removed**

- n/a

#### Testing / Reviewing
**NOTE: InlineCheckbox is most easily tested as a child of `TableSelectAll`
1. Open Storybook > DataTable > Selection > Default in browser for Dev Tool testing
2. Open `TableSelectAll.tsx` in code
3. Add one aria label type at a time to the component in story code. 
 - 1. Only one version can be active at a time.
   - `aria-label` (new)
   - `ariaLabel` (deprecated))
 - 2. Be sure to find the correct usage for the given component as explained in code or on the story itself
 - 3. Test examples below
   - Note: For my testing, I duplicate the lines and then comment out the one I am not using. The image blow does NOT yet comment out one of the test lines:
![image](https://user-images.githubusercontent.com/9935383/222556612-55260bde-0244-4003-8bda-84a1c2b68a7a.png)


4. confirm that both propTypes still display in Dev Tools as intended
5. this is NOT on the Component API for any DataTable items and cannot be confirmed
